tests/model/model_flattening.cpp

```diff
@@ -0,0 +1,623 @@
+        "  <component name=\"best_component\">\n"
+        "    <variable name=\"v\" units=\"common_units\" initial_value=\"1\" interface=\"public_and_private\"/>\n"
+        "  </component>\n"
+        "</model>\n";
+
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("modelflattening/importedunitswithnameclashes.xml"));
+
+    EXPECT_TRUE(model->hasUnresolvedImports());
+    model->resolveImports(resourcePath("modelflattening/"));
+    EXPECT_FALSE(model->hasUnresolvedImports());
+
+    model->flatten();
+
+    auto printer = libcellml::Printer::create();
+
+    auto a = printer->printModel(model);
+    EXPECT_EQ(e, a);
+}
+
+TEST(ModelFlattening, importedComponentWithNameClashes)
+{
+    const std::string e =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"main_model\">\n"
+        "  <units name=\"common_units\">\n"
+        "    <unit units=\"second\"/>\n"
+        "  </units>\n"
+        "  <component name=\"my_component\"/>\n"
+        "  <component name=\"best_component_1\"/>\n"
+        "  <component name=\"best_component\">\n"
+        "    <variable name=\"v\" units=\"common_units\" initial_value=\"1\" interface=\"public_and_private\"/>\n"
+        "  </component>\n"
+        "  <encapsulation>\n"
+        "    <component_ref component=\"my_component\">\n"
+        "      <component_ref component=\"best_component_1\"/>\n"
+        "    </component_ref>\n"
+        "  </encapsulation>\n"
+        "</model>\n";
+
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("modelflattening/importedcomponentswithnameclashes.xml"));
+
+    EXPECT_TRUE(model->hasUnresolvedImports());
+    model->resolveImports(resourcePath("modelflattening/"));
+    EXPECT_FALSE(model->hasUnresolvedImports());
+
+    model->flatten();
+
+    auto printer = libcellml::Printer::create();
+
+    auto a = printer->printModel(model);
+    EXPECT_EQ(e, a);
+}
+
+// The variable ordering changes when using the HH model defined with imports.
+TEST(ModelFlattening, hodgkinHuxleyDefinedUsingImports)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("modelflattening/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+
+    EXPECT_TRUE(model->hasUnresolvedImports());
+    model->resolveImports(resourcePath("modelflattening/hodgkin_huxley_squid_axon_model_1952/"));
+    EXPECT_FALSE(model->hasUnresolvedImports());
+
+    model->flatten();
+
+    auto printer = libcellml::Printer::create();
+
+    auto a = printer->printModel(model);
+    auto modelNonImportVersion = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+    auto e = printer->printModel(modelNonImportVersion);
+    // Sadly cannot get the order of the connections to match each other, everything else does.
+    // EXPECT_EQ(e, a);
```

Comment:
I am going to reenable that check since we clearly want it. I am just going to double check the "original" HH52 model and modify accordingly, if it is acceptable.

Reply:
Done.